include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,453 @@
+
+#ifndef SEQAN_INCLUDE_ARG_PARSE_VERSION_CHECK_H_
+#define SEQAN_INCLUDE_ARG_PARSE_VERSION_CHECK_H_
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <sys/stat.h>
+#include <time.h>
+#include <stdio.h>
+#include <regex>
+#include <future>
+#include <sys/types.h> 
+#include <chrono>
+
+namespace seqan
+{
+
+// ==========================================================================
+// Forwards
+// ==========================================================================
+
+// NOTE(rrahn): In-file forward for function call operator.
+struct VersionCheck;
+inline void _checkForNewerVersion(VersionCheck &, std::promise<bool>);
+inline std::string _getOS();
+inline std::string _getPath();
+inline std::string _getBitSys();
+
+// ==========================================================================
+// Tags, Classes, Enums
+// ==========================================================================
+
+template <typename TVoidSpec = void>
+struct VersionControlTags_
+{
+    static constexpr char const * const SEQAN_NAME = "seqan";
+    static constexpr char const * const UNREGISTERED_APP= "UNREGISTERED APP";
+    static constexpr char const * const OPTION_OFF = "OFF";
+    static constexpr char const * const OPTION_DEV = "DEV";
+    static constexpr char const * const OPTION_APP_ONLY = "APP_ONLY";
+};
+
+template <typename TVoidSpec>
+constexpr char const * const VersionControlTags_<TVoidSpec>::SEQAN_NAME;
+template <typename TVoidSpec>
+constexpr char const * const VersionControlTags_<TVoidSpec>::UNREGISTERED_APP;
+template <typename TVoidSpec>
+constexpr char const * const VersionControlTags_<TVoidSpec>::OPTION_OFF;
+template <typename TVoidSpec>
+constexpr char const * const VersionControlTags_<TVoidSpec>::OPTION_DEV;
+template <typename TVoidSpec>
+constexpr char const * const VersionControlTags_<TVoidSpec>::OPTION_APP_ONLY;
+
+struct VersionCheck
+{
+    // ----------------------------------------------------------------------------
+    // Member Variables
+    // ----------------------------------------------------------------------------
+    std::string _url;
+    std::string _name;
+    std::string _version = "0.0.0";
+    std::string _program;
+    std::string _command;
+    std::string _website = "https://github.com/seqan/seqan/tree/master/apps";
+    std::string _path = _getPath();
+
+    // ----------------------------------------------------------------------------
+    // Constructors
+    // ----------------------------------------------------------------------------
+
+    VersionCheck(std::string const & name,
+                 std::string const & version,
+                 std::string const & website)
+    {
+        _name = name;
+        if (!version.empty() &&
+            std::regex_match(version, std::regex("^[[:digit:]]+\\.[[:digit:]]+\\.[[:digit:]]+.*")))
+            _version = version.substr(0,5); // in case the git revision number is given take only version number
```

Comment:
what if the version is `10.23.12` ? then `substr()` will not get it all... Rather extract it with the regex module directly...